### PR TITLE
Native Inbox Reply

### PR DESCRIPTION
**Pull request by domonkosadam**
refs: MBL-14852
affects: Student, Teacher
release note: none

test plan: Steps are in the ticket

## Screenshots

<table>
<tr><th>Before</th><th>After</th></tr>
<tr>
<td></td>
<td></td>
</tr>
</table>

## Checklist

- [ ] A11y checked
- [ ] Tested on phone
- [ ] Tested on tablet
- [ ] Tested in dark mode
- [ ] Tested in light mode

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Build failed, skipping coverage check</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This pull request will not generate a release note.</td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Teacher, Student

[MBL-14852](https://instructure.atlassian.net/browse/MBL-14852)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e01b1ebf252c9ae571670998407626c5f1bbb174
</p>

**Comment by inst-danger** [comment]
<details><summary>Teacher Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

File: Core/Core/Inbox/ComposeMessage/ViewModel/ComposeMessageViewModel.swift [hunk 1]
```diff
@@ -35,29 +33,56 @@ class ComposeMessageViewModel: ObservableObject {
...
+            self.subject = conversation.subject
+            let rawContext = conversation.contextCode?.split(separator: "_") ?? []
+            switch rawContext[0] {
+            case "course":
+                self.selectedContext = RecipientContext(name: conversation.contextName ?? "", context: Context.course(String(rawContext[1])))
+            case "group":
+                self.selectedContext = RecipientContext(name: conversation.contextName ?? "", context: Context.group(String(rawContext[1])))
+            default:
+                self.selectedContext = RecipientContext(name: conversation.contextName ?? "", context: Context.course(String(rawContext[1])))
+            }
```
Comment (vargaat) [inline Core/Core/Inbox/ComposeMessage/ViewModel/ComposeMessageViewModel.swift:75]: I think we could use `Context(canvasContextID: String)` to make this snippet more simple.

File: Core/Core/Inbox/MessageDetails/Model/MessageDetailsInteratorLive.swift [hunk 2]
```diff
@@ -44,6 +45,9 @@ public class MessageDetailsInteractorLive: MessageDetailsInteractor {
 
         conversationStore
             .allObjects
+            .handleEvents(receiveOutput: { conversation in
+                self.conversation.value = conversation
+            })
```
Comment (vargaat) [inline Core/Core/Inbox/MessageDetails/Model/MessageDetailsInteratorLive.swift:50]: I think it would be clearer to setup a new stream for this.

**Review by vargaat — CHANGES_REQUESTED** [review]
QA+1, I had a few code review findings but otherwise looks good!

**Review by szabinst — APPROVED** [review]
QA + 1 

The send button changes position when I type more than 2 characters. Let's not forget fixing it in the next PR.

**Pull request merged**